internal/platform/file.go

```diff
@@ -0,0 +1,50 @@
+package platform
+
+import (
+	"io"
+	"io/fs"
+)
+
+// ReadFile declares all read interfaces defined on os.File used by wazero.
```

Comment:
these types were moved from sysfs package, and yeah soon we'll swap these out for our own file type. nearly there.